doc/tutorials/ml/introduction_to_pca/introduction_to_pca.markdown

```diff
@@ -0,0 +1,151 @@
+
+Goal
+----
+
+In this tutorial you will learn how to:
+
+-   Use the OpenCV class @ref PCA to calculate the orientation of an object.
+
+What is PCA?
+--------------
+
+Principal Component Analysis (PCA) is a statistical procedure that extracts the most important features of a dataset.
+
+<p>![](images/pca_line.png)</p>
+
+Consider that you have a set of 2D points as it is shown in the figure above. Each dimension corresponds to a feature you are interested in. Here some could argue that the points are set in a random order. However, if you have a better look you will see that there is a linear pattern (indicated by the blue line) which is hard to dismiss. A key point of PCA is the Dimensionality Reduction. Dimensionality Reduction is the process of reducing the number of the dimensions of the given dataset. For example, in the above case it is possible to approximate the set of points to a single line and therefore, reduce the dimensionality of the given points from 2D to 1D.
+
+Moreover, you could also see that the points vary the most along the blue line, more than they vary along the Feature 1 or Feature 2 axes. This means that if you know the position of a point along the blue line you have more information about the point than if you only knew where it was on Feature 1 axis or Feature 2 axis.
+
+Hence, PCA allows us to find the direction along which our data varies the most. In fact, the result of running PCA on the set of points in the diagram consist of 2 vectors called _eigenvectors_ which are the _principal components_ of the data set.
+
+<p>![](images/pca_eigen.png)</p>
+
+The size of each eigenvector is encoded in the corresponding eigenvalue and indicates how much the data vary along the principal component. The beginning of the eigenvectors is the center of all points in the data set. Applying PCA to N-dimensional data set yields N N-dimensional eigenvectors, N eigenvalues and 1 N-dimensional center point. Enough theory, let’s see how we can put these ideas into code.
+
+How are the eigenvectors and eigenvalues computed?
+--------------------------------------------------
+
+<p>The goal is to transform a given data set <strong>X</strong> of dimension <em>p</em> to an alternative data set <strong>Y</strong> of smaller dimension <em>L</em>. Equivalently, we are seeking to find the matrix <strong>Y</strong>, where <strong>Y</strong> is the _Karhunen–Loève transform_ (KLT) of matrix <strong>X</strong>:</p>
+
+<p>![](images/klt.png)</p>
+
+<p><strong>Organize the data set</strong></p>
+
+<p>Suppose you have data comprising a set of observations of <em>p</em> variables, and you want to reduce the data so that each observation can be described with only <em>L</em> variables, <em>L</em> &lt; <em>p</em>. Suppose further, that the data are arranged as a set of <em>n</em> data vectors ![](images/xdotsx.png) with each ![](images/xi.png) representing a single grouped observation of the <em>p</em> variables.</p>
```

Comment:
Here you tried to insert inline formula, but doxygen does not insert images inline. Use `\f$ x_1...x_n \f$` instead of `xdotsx.png` and `\f$ x_i \f$` instead of `xi.png`.